spring/spring4/spring4-common/src/test/java/org/infinispan/spring/session/InfinispanApplicationPublishedBridgeTCK.java

```diff
@@ -0,0 +1,79 @@
+package org.infinispan.spring.session;
+
+import static com.sun.corba.se.impl.util.RepositoryId.cache;
+import static org.testng.Assert.assertNull;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.infinispan.spring.provider.SpringCache;
+import org.infinispan.spring.session.util.EventsWaiter;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.session.MapSession;
+import org.springframework.session.events.SessionCreatedEvent;
+import org.springframework.session.events.SessionDeletedEvent;
+import org.springframework.session.events.SessionExpiredEvent;
+import org.testng.annotations.Test;
+
+public abstract class InfinispanApplicationPublishedBridgeTCK {
+
+   protected SpringCache springCache;
+
+   protected AbstractInfinispanSessionRepository sessionRepository;
+
+   protected abstract SpringCache createSpringCache();
+
+   protected abstract void callEviction();
+
+   protected abstract AbstractInfinispanSessionRepository createRepository(SpringCache springCache) throws Exception;
+
+   protected void init() throws Exception {
+      springCache = createSpringCache();
+      sessionRepository = createRepository(springCache);
+   }
+
+   @Test
+   public void testEventBridge() throws Exception {
+      //given
+      EventsCollector eventsCollector = new EventsCollector();
+
+      sessionRepository.setApplicationEventPublisher(eventsCollector);
+
+      //when
+      MapSession sessionToBeDeleted = sessionRepository.createSession();
+      MapSession sessionToBeExpired = sessionRepository.createSession();
+      sessionToBeExpired.setMaxInactiveIntervalInSeconds(1);
+
+      sessionRepository.save(sessionToBeExpired);
+      sessionRepository.save(sessionToBeDeleted);
+      sessionRepository.delete(sessionToBeDeleted.getId());
+
+      TimeUnit.SECONDS.sleep(1);
+      callEviction();
+
+      //then
+      assertNull(cache.get(sessionToBeExpired.getId()));
+      assertNull(cache.get(sessionToBeDeleted.getId()));
+      EventsWaiter.assertNumberOfEvents(() -> eventsCollector.getEvents(), SessionCreatedEvent.class, 2, 2, TimeUnit.SECONDS);
+      EventsWaiter.assertNumberOfEvents(() -> eventsCollector.getEvents(), SessionDeletedEvent.class, 1, 2, TimeUnit.SECONDS);
+      EventsWaiter.assertNumberOfEvents(() -> eventsCollector.getEvents(), SessionExpiredEvent.class, 1, 2, TimeUnit.SECONDS);
```

Comment:
Yes, we need to count items fulfilling certain condition in the stream of events. I would share a `List` here but I though using a `Stream` with supplier would be nicer..